src/tests/jsonLogic.test.js

```diff
@@ -0,0 +1,213 @@
+      const { formErrors } = handleValidation({ field_a: 1, field_b: 2 });
+      expect(formErrors.field_a).toEqual('Field A must be bigger than field B');
+      expect(handleValidation({ field_a: 2, field_b: 0 }).formErrors).toEqual(undefined);
+    });
+
+    it('smaller: field_a < field_b', () => {
+      const schema = createSchemaWithRulesOnFieldA({
+        a_less_than_b: {
+          errorMessage: 'Field A must be smaller than field B',
+          rule: { '<': [{ var: 'field_a' }, { var: 'field_b' }] },
+        },
+      });
+      const { handleValidation } = createHeadlessForm(schema, { strictInputType: false });
+      const { formErrors } = handleValidation({ field_a: 2, field_b: 2 });
+      expect(formErrors.field_a).toEqual('Field A must be smaller than field B');
+      expect(handleValidation({ field_a: 0, field_b: 2 }).formErrors).toEqual(undefined);
+    });
+
+    it('equal: field_a = field_b', () => {
+      const schema = createSchemaWithRulesOnFieldA({
+        a_equals_b: {
+          errorMessage: 'Field A must equal field B',
+          rule: { '==': [{ var: 'field_a' }, { var: 'field_b' }] },
+        },
+      });
+      const { handleValidation } = createHeadlessForm(schema, { strictInputType: false });
+      const { formErrors } = handleValidation({ field_a: 3, field_b: 2 });
+      expect(formErrors.field_a).toEqual('Field A must equal field B');
+      expect(handleValidation({ field_a: 2, field_b: 2 }).formErrors).toEqual(undefined);
+    });
+  });
+
+  describe('Arithmetic: +, -, *, /', () => {
+    it('multiple: field_a > field_b * 2', () => {
+      const schema = createSchemaWithRulesOnFieldA({
+        a_greater_than_b_multiplied_by_2: {
+          errorMessage: 'Field A must be at least twice as big as field b',
+          rule: { '>': [{ var: 'field_a' }, { '*': [{ var: 'field_b' }, 2] }] },
+        },
+      });
+      const { handleValidation } = createHeadlessForm(schema, { strictInputType: false });
+
+      const { formErrors } = handleValidation({ field_a: 1, field_b: 4 });
+      expect(formErrors.field_a).toEqual('Field A must be at least twice as big as field b');
+      expect(handleValidation({ field_a: 3, field_b: 1 }).formErrors).toEqual(undefined);
+    });
+
+    it('divide: field_a > field_b / 2', () => {
+      const { handleValidation } = createHeadlessForm(
+        createSchemaWithRulesOnFieldA({
+          a_greater_than_b_divided_by_2: {
+            errorMessage: 'Field A must be greater than field_b / 2',
+            rule: { '>': [{ var: 'field_a' }, { '/': [{ var: 'field_b' }, 2] }] },
+          },
+        }),
+        { strictInputType: false }
+      );
+      const { formErrors } = handleValidation({ field_a: 2, field_b: 4 });
+      expect(formErrors.field_a).toEqual('Field A must be greater than field_b / 2');
+      expect(handleValidation({ field_a: 3, field_b: 5 }).formErrors).toEqual(undefined);
```

Comment:
nit: Why do we store `formErrors` above but here it's inline? Make it all inline! 😅 Or use the `validateForm` helper that is used in the other tests

Reply:
this was like one of the earliest tests and i didnt know any better 😂

Reply:
fixed